tests/rkt_fetch_test.go

```diff
@@ -0,0 +1,198 @@
+// Copyright 2015 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"strings"
+	"testing"
+
+	"github.com/coreos/rkt/Godeps/_workspace/src/github.com/ThomasRooney/gexpect"
+)
+
+// TestFetch tests that 'rkt fetch' will always bypass the on-disk store and
+// fetch from a local file or from a remote URL.
+func TestFetch(t *testing.T) {
+	image := "rkt-inspect-fetch.aci"
+
+	ctx := newRktRunCtx()
+	defer ctx.cleanup()
+
+	// Fetch the image for the first time, this should write the image to the
+	// on-disk store.
+
+	patchTestACI(image, "--exec=/inspect --read-file")
+
+	oldHash := importImageAndFetchHash(t, ctx, image)
+	os.Remove(image)
+
+	// Fetch the image with the same name but different content, the expecting
+	// result is that we should get a different hash since we are not fetching
+	// from the on-disk store.
+
+	patchTestACI(image, "--exec=/inspect --read-file --write-file")
+
+	newHash := importImageAndFetchHash(t, ctx, image)
+	os.Remove(image)
+
+	if oldHash == newHash {
+		t.Fatalf("ACI hash should be different as the image has changed")
+	}
+}
+
+// TestRunPrepareFromFile tests that when 'rkt run/prepare' a local ACI, it will bypass the
+// on-disk store.
+func TestRunPrepareFromFile(t *testing.T) {
+	foundMsg := "found image in local store"
+	image := "rkt-inspect-implicit-fetch.aci"
+
+	patchTestACI(image, "--exec=/inspect")
+	defer os.Remove(image)
+
+	pwd, err := os.Getwd()
+	if err != nil {
+		t.Fatalf("Cannot get current working directory: %v", err)
```

Comment:
Well the way I like to distinguish these is that Error/Fatal are for actual test case failures and Panic is for setup/runtime issues that indicate unexpected things unrelated to expected test behaviour per se